vent/common/utils.py

```diff
@@ -0,0 +1,34 @@
+import signal
+import time
+from contextlib import contextmanager
+
+class TimeoutException(Exception): pass
+
+@contextmanager
+def time_limit(seconds):
+    def signal_handler(signum, frame):
+        raise TimeoutException("Timed out!")
+    signal.signal(signal.SIGALRM, signal_handler)
+    signal.setitimer(signal.ITIMER_REAL, seconds)
+    try:
+        yield
```

Comment:
Can this handle returning from a called method?

Reply:
I thought so; why not? I can write a test for that

Reply:
wrote test, included in my commit :)

Reply:
awesome, thanks!